tools/base/libboard/board_vc/board_vc.c

```diff
@@ -0,0 +1,749 @@
+	fpga_result resval              = FPGA_OK;
+	char ver_a_buf[VER_BUF_SIZE]    = { 0 };
+	char ver_b_buf[VER_BUF_SIZE]    = { 0 };
+	uint32_t size                   = 0;
+	fpga_object pkvl_a_object;
+	fpga_object pkvl_b_object;
+
+
+	res = fpgaTokenGetObject(token, SYSFS_PKVL_A_VER, &pkvl_a_object, FPGA_OBJECT_GLOB);
+	if (res != FPGA_OK) {
+		OPAE_MSG("Failed to get token object");
+		return res;
+	}
+
+	res = fpgaTokenGetObject(token, SYSFS_PKVL_B_VER, &pkvl_b_object, FPGA_OBJECT_GLOB);
+	if (res != FPGA_OK) {
+		OPAE_MSG("Failed to get token object");
+		resval = res;
+		goto out_destroy_obj_a;
+	}
+
+
+	res = fpgaObjectGetSize(pkvl_a_object, &size, FPGA_OBJECT_GLOB);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to get object size");
+		resval = res;
+		goto out_destroy_obj_b;
+	}
+
+	res = fpgaObjectRead(pkvl_a_object, (uint8_t *)ver_a_buf, 0, size, 0);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to read object ");
+		resval = res;
+		goto out_destroy_obj_b;
+	}
+
+	res = fpgaObjectGetSize(pkvl_b_object, &size, FPGA_OBJECT_GLOB);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to get object size");
+		resval = res;
+		goto out_destroy_obj_b;
+	}
+
+	res = fpgaObjectRead(pkvl_b_object, (uint8_t *)ver_b_buf, 0, size, 0);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to read object ");
+		resval = res;
+		goto out_destroy_obj_b;
+	}
+
+	printf("%-32s : %s", "Retimer A Version", ver_a_buf);
+	printf("%-32s : %s", "Retimer B Version", ver_b_buf);
+
+out_destroy_obj_b:
+	res = fpgaDestroyObject(&pkvl_b_object);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to destroy object");
+	}
+
+out_destroy_obj_a:
+	res = fpgaDestroyObject(&pkvl_a_object);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to destroy object");
+	}
+
+	return resval;
+}
+
+// print mac information
+fpga_result print_mac_info(fpga_token token)
+{
+	fpga_result res               = FPGA_OK;
+	unsigned char buf[8]          = { 0 };
+	int i                         = 0;
+	int n                         = 0;
+	pkvl_mac mac;
+
+	res = read_mac_info(token, buf, SYSFS_MAX_SIZE);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to read mac information");
+		return res;
+	}
+
+	n = (int)buf[6];
+	printf("%-32s : %d\n", "Number of MACs", n);
+	mac.byte[0] = buf[5];
+	mac.byte[1] = buf[4];
+	mac.byte[2] = buf[3];
+	mac.byte[3] = 0;
+	for (i = 0; i < n; ++i) {
+		printf("%s %-20d : %02X:%02X:%02X:%02X:%02X:%02X\n",
+			"MAC address", i, buf[0], buf[1], buf[2],
+			mac.byte[2], mac.byte[1], mac.byte[0]);
+		mac.dword += 1;
+	}
+
+	return res;
+}
+
+// print board information
+fpga_result print_board_info(fpga_token token)
+{
+	fpga_result res                      = FPGA_OK;
+	char bmc_ver[SYSFS_MAX_SIZE]         = { 0 };
+	char max10_ver[SYSFS_MAX_SIZE]       = { 0 };
+	char pcb_ver[SYSFS_MAX_SIZE]         = { 0 };
+
+	res = read_bmcfw_version(token, bmc_ver, SYSFS_MAX_SIZE);
+	if (res != FPGA_OK) {
+		OPAE_MSG("Failed to read bmc version");
+	}
+
+	res = read_max10fw_version(token, max10_ver, SYSFS_MAX_SIZE);
```

Comment:
`SYSFS_MAX_SIZE` is used eventually when calling `snprintf_s_ciii` as the limit of the buffer being written to (similar to above).

Reply:
Fixed